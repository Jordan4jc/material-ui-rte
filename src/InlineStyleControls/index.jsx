import React from "react"
import { makeStyles } from "@material-ui/core/styles"
import Button from "@material-ui/core/Button"
import ButtonGroup from "@material-ui/core/ButtonGroup"
import FormatBoldIcon from "@material-ui/icons/FormatBold"
import FormatItalicIcon from "@material-ui/icons/FormatItalic"
import FormatUnderlinedIcon from "@material-ui/icons/FormatUnderlined"
import clsx from "clsx"
import styles from "./styles"

const useStyles = makeStyles(styles, {
  name: "MuiRte-InlineStyleControls"
})

const INLINE_STYLES = [
  { icon: FormatBoldIcon, style: "BOLD" },
  { icon: FormatItalicIcon, style: "ITALIC" },
  { icon: FormatUnderlinedIcon, style: "UNDERLINE" }
]

const InlineStyleControls = props => {
  const classes = useStyles()
  const currentStyle = props.editorState.getCurrentInlineStyle()
  const renderIcon = type => {
    let IconName = type.icon
    return (
      <IconName
        className={clsx({
          [classes.activeIcon]: currentStyle.has(type.style)
        })}
      />
    )
  }
  return (
    <ButtonGroup size="small" variant="contained">
      {INLINE_STYLES.map(type => (
        <Button key={type.style} onClick={() => props.onChange(type.style)}>
          {renderIcon(type)}
        </Button>
      ))}
    </ButtonGroup>
  )
}

export default InlineStyleControls
