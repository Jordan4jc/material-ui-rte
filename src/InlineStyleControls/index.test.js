import React from "react"
import { shallow, mount } from "enzyme"
import { EditorState } from "draft-js"
import InlineStyleControls from "./index"
import Button from "@material-ui/core/Button"

describe("InlineStyleControls", () => {
  let editorState

  beforeEach(() => {
    editorState = EditorState.createEmpty()
  })

  it("Should render block style control", () => {
    const component = shallow(<InlineStyleControls editorState={editorState} />)

    expect(component).toMatchSnapshot()
  })

  it("Should only render passed inline style buttons", () => {
    const controls = ["bold", "italic"]
    const component = mount(
      <InlineStyleControls editorState={editorState} controls={controls} />
    )

    const renderedButtons = component.find(Button)

    expect(renderedButtons.length).toBe(2)
    expect(renderedButtons.at(0).key()).toBe(".$BOLD")
    expect(renderedButtons.at(1).key()).toBe(".$ITALIC")
  })
})
