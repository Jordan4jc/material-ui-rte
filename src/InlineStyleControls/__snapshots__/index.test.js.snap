// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineStyleControls Should render block style control 1`] = `
<WithStyles(ForwardRef(ButtonGroup))>
  <WithStyles(ForwardRef(Button))
    variant="contained"
  >
    <FormatBoldIcon />
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    variant="contained"
  >
    <FormatItalicIcon />
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    variant="contained"
  >
    <FormatUnderlinedIcon />
  </WithStyles(ForwardRef(Button))>
</WithStyles(ForwardRef(ButtonGroup))>
`;
